core/lib/eth_client/src/types.rs

```diff
@@ -129,6 +134,64 @@ pub struct SignedCallResult {
     pub hash: H256,
 }
 
+impl SignedCallResult {
+    pub fn new(
+        raw_tx: RawTransactionBytes,
+        max_priority_fee_per_gas: U256,
+        max_fee_per_gas: U256,
+        nonce: U256,
+        hash: H256,
+    ) -> Self {
+        Self {
+            raw_tx,
+            max_priority_fee_per_gas,
+            max_fee_per_gas,
+            nonce,
+            hash,
+        }
+    }
+
+    pub fn raw_tx(&self, blob_tx_sidecar: Option<&EthTxBlobSidecar>) -> RawTransactionBytes {
+        match blob_tx_sidecar {
+            None => self.raw_tx.clone(),
+            Some(sidecar) => encode_blob_tx_with_sidecar(&self.raw_tx, sidecar),
+        }
+    }
+}
+
+fn encode_blob_tx_with_sidecar(
```

Comment:
can you add a simple 'smoke' unittest that checks the output of more than 1 blob.

it can be as simple as keccak(encode_blob_tx(sidecar=vec['data1', 'data2'])) == 0x1234

so that if we ever change this logic by mistake, the test would fail.